test/metabase/api/dashboard_test.clj

```diff
@@ -383,7 +385,20 @@
                                             :cache_ttl     1234
                                             :creator_id    (mt/user->id :rasta)
                                             :collection_id true})
-                 (dashboard-response (db/select-one Dashboard :id dashboard-id)))))))))
+                 (dashboard-response (db/select-one Dashboard :id dashboard-id)))))
+
+        (testing "No-op PUT: Do not return 500"
+          (t2.with-temp/with-temp [Card          {card-id :id} {}
+                                   DashboardCard dashcard      {:card_id card-id, :dashboard_id dashboard-id}]
+            ;; so, you can't actually set `:cards` with THIS endpoint (you have to use PUT /api/dashboard/:id/cards),
+            ;; but the e2e tests are trying to do it. With Toucan 1, it would silently do nothing and return truthy for
+            ;; whatever reason (I'm guessing it was a bug?) if you did something like (update! Dashboard 1 {}). Toucan 2
+            ;; returns falsey, since it doesn't do anything, which is what Toucan 1 SAID it was supposed to do.
+            ;;
+            ;; In the interest of un-busting the e2e tests let's just check to make sure the endpoint no-ops
+            (is (=? {:id dashboard-id}
+                    (mt/user-http-request :rasta :put 200 (str "dashboard/" dashboard-id)
+                                          {:cards [(select-keys dashcard [:id :card_id :row_col :size_x :size_y])]})))))))))
```

Comment:
This was the thing that was causing e2e tests to barf. we didn't have a test around this. Toucan 2 didn't have the bug that Toucan1 had so it started erroring on this.

To be fair, this is a broken request, but fixing the e2e tests doing broken things is not something I want to deal with in this PR